midi_app_controller/state/state_manager.py

```diff
@@ -195,3 +206,26 @@ def start_handling(self) -> None:
             midi_in=self._midi_in,
             midi_out=self._midi_out,
         )
+
+    def save_state(self):
+        AppState(
+            selected_controller_path=self.selected_controller.path
+            if self.selected_controller
+            else None,
+            selected_binds_path=self.selected_binds.path
+            if self.selected_binds
+            else None,
+            selected_midi_in=self.selected_midi_in,
+            selected_midi_out=self.selected_midi_out,
+            recent_binds_for_controller=self.recent_binds_for_controller,
+        ).save_to(Config.APP_STATE_FILE)
+
+    def load_state(self):
+        if not Config.APP_STATE_FILE.exists():
+            return
+        state = AppState.load_from(Config.APP_STATE_FILE)
+        self.select_controller(state.selected_controller_path)
+        self.select_binds(state.selected_binds_path)
+        self.select_midi_in(state.selected_midi_in)
+        self.select_midi_out(state.selected_midi_out)
+        self.recent_binds_for_controller = state.recent_binds_for_controller
```

Comment:
I think that it should be verified if the paths are inside `Config.BINDS_READONLY_DIR` etc. (here or in `AppState`). If a path is not inside it, it should be ignored.

For example if someone opens napari in a new `venv`, paths to config files in the old `venv` would be used. And I don't think we want it.